app/lib/csv_parser.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+class CSVParser
+  def initialize(data)
+    @data = data
+  end
+
+  def call
+    CSV.parse(
+      data,
+      converters:,
+      empty_value: nil,
+      encoding:,
+      header_converters:,
+      headers: true,
+      skip_blanks: true,
+      strip: true
+    )
+  end
+
+  def self.call(...) = new(...).call
+
+  private_class_method :new
+
+  Field =
+    Struct.new("Field", :value, :column, :row, :header) do
+      delegate :blank?, :present?, to: :value
+
+      alias_method :to_s, :value
+
+      def cell = "#{column}#{row}"
+
+      def to_i
+        Integer(value)
+      rescue ArgumentError, TypeError
+        nil
+      end
+
+      def to_date
+        return nil if blank?
+
+        parsed_values =
+          DATE_FORMATS.lazy.filter_map do |format|
+            Date.strptime(value, format)
+          rescue ArgumentError, TypeError
+            nil
+          end
+
+        parsed_values.first
+      end
+
+      def to_postcode
+        if present?
+          postcode = UKPostcode.parse(value)
+          postcode.to_s if postcode.full_valid?
+        end
+      end
+
+      def to_time
+        return nil if blank?
+
+        parsed_values =
+          TIME_FORMATS.lazy.filter_map do |format|
+            Time.zone.strptime(value, format)
+          rescue ArgumentError, TypeError
+            nil
+          end
+
+        parsed_values.first
+      end
+    end
+
+  private
+
+  attr_reader :data
+
+  ALPHABET = %w[A B C D E F G H I J K L M N O P Q R S T U V W X Y Z].freeze
```

Comment:
```suggestion
  ALPHABET =  ('A'..'Z').to_a.freeze
  ALPHABET + ALPHABET.product(ALPHABET).map(&:join)
```

Reply:
Ah nice! I've amended the commit to do this instead.